file-formats/sajt/type/zif_aff_sajt_v1.intf.abap

```diff
@@ -0,0 +1,103 @@
+INTERFACE zif_aff_sajt_v1
+  PUBLIC.
+
+  TYPES:
+    "! <p class="shorttext">Header for SAJT Object</p>
+    "! The header for an application job template
+    BEGIN OF ty_header,
+      "! <p class="shorttext">Description</p>
+      "! Description of the application job template
+      "! $required
+      description           TYPE c LENGTH 120,
+      "! <p class="shorttext">Original Language</p>
+      "! Original language of the application job template
+      "! $required
+      original_language     TYPE sy-langu,
+      "! <p class="shorttext">ABAP Language Version</p>
+      "! ABAP language version
+      "! $values {@link zif_aff_types_v1.data:co_abap_language_version_cloud}
+      "! $default {@link zif_aff_types_v1.data:co_abap_language_version_cloud.standard}
+      abap_language_version TYPE zif_aff_types_v1=>ty_abap_language_version,
+    END OF ty_header.
+
+  TYPES:
+    "! <p class="shorttext">Application Job Catalog Entry</p>
+    "! Name of the application job catalog entry
+    ty_catalog_name TYPE c LENGTH 40.
+
+  "! <p class="shorttext">Sign</p>
+  "! Include/exclude values (I/E)
+  "! $values {@link zif_aff_sajt_v1.data:co_sign}
+  TYPES ty_sign TYPE c LENGTH 1.
+
+  CONSTANTS:
+    "! <p class="shorttext">Sign</p>
+    "! Include/exclude values (I/E)
+    BEGIN OF co_sign,
+      "! <p class="shorttext">Exclude</p>
+      "! Exclude values
+      exclude TYPE ty_sign VALUE 'E',
+      "! <p class="shorttext">Include</p>
+      "! Include values
+      include TYPE ty_sign VALUE 'I',
+    END OF co_sign.
+
+  TYPES:
+    "! <p class="shorttext">Entry of Ranges Table</p>
+    "! Entry of ranges table
+    BEGIN OF ty_value_range_entry,
+      "! <p class="shorttext">Sign</p>
+      "! Include/exclude values (I/E)
+      sign      TYPE  ty_sign,
+      "! <p class="shorttext">Option</p>
+      "! Operator of the ranges condition (EQ/NE/GE/GT/LE/LT/CP/NP/BT/NB)
+      option    TYPE  zif_aff_types_v1=>ty_option,
+      "! <p class="shorttext">Low Value</p>
+      "! Low value of the ranges condition
+      low       TYPE  c LENGTH 255,
+      "! <p class="shorttext">High Value</p>
+      "! High value of the ranges condition
+      high      TYPE  c LENGTH 255,
+    END OF ty_value_range_entry.
+
+  TYPES:
+    "! <p class="shorttext">Parameter Values</p>
+    "! Parameter values as ranges table
+    ty_value_ranges_table TYPE STANDARD TABLE OF ty_value_range_entry WITH EMPTY KEY.
```

Comment:
We suggest to use `DEFAULT KEY` instead of `EMPTY KEY`

Reply:
we should tell this to abaplint  in the abaplint.jsonc file

Reply:
> we should tell this to abaplint in the abaplint.jsonc file

I think abaplint and our configuration is fine. We have to change it as suggested by abaplint, because the addition `EMPTY KEY` is only supported in releases 7.40 and higher :(